streams/src/test/java/org/apache/kafka/streams/state/internals/InMemoryLRUCacheStoreTest.java

```diff
@@ -48,6 +54,23 @@
     }
 
     @Test
+    public void shouldPutAll() {
```

Comment:
I'd split this into two tests. Something like:
`shouldPutAllKeyValuePairs` it should also check that the correct values were inserted. So it might look something like this:
```
final List<KeyValue<Integer, String>> kvPairs = Arrays.asList(KeyValue.pair(1, "1"),
                                                                      KeyValue.pair(2, "2"),
                                                                      KeyValue.pair(3, "3"));
        store.putAll(kvPairs);

        assertThat(store.approximateNumEntries(), equalTo(3L));
        for (KeyValue<Integer, String> kvPair : kvPairs) {
            assertThat(store.get(kvPair.key), equalTo(kvPair.value));
        }
```

I'd also add another test that shows that you can update the kvPairs and the size remains the same, but the value for any key is the new value. `shouldUpdateValuesForExistingKeysOnPutAll`
So it would be fairly similar to above, but with 2 calls to `putAll`